src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -82,6 +86,46 @@ case class WorkflowActor(workflow: WorkflowDescriptor,
     Await.result(futureStore, DatabaseTimeout)
   }
 
+ /**
+   * Try to generate output for a collector call, by collecting outputs for all of its shards.
+   * It's fail-fast on shard output retrieval
+   */
+  private def generateCollectorOutput(collector: CollectorKey, shards: Iterable[CallKey]): Try[CallOutputs] = Try {
+   /* FIXME: use _.index.fromIndex when merged to "Database SG branch" */
+    val shardsOutputs = shards.toSeq sortBy { _.index.fromIndex } map { e =>
+      fetchCallOutputEntries(e) map { _.value } get
+    }
+   /* FIXME: No need for asInstanceOf[Call] when merged to "Database SG branch" */
+    collector.scope.task.outputs map { taskOutput =>
+      val wdlValues = shardsOutputs.map(_.get(taskOutput.name).get)
+      taskOutput.name -> new WdlArray(WdlArrayType(taskOutput.wdlType), wdlValues)
+    } toMap
+  }
+
+  private def findShards(key: CollectorKey) = executionStore collect {
+    case (k: CallKey, _) if k.scope == key.scope && isShard(k) => k
+  }
+
+  /**
+   * Gathers and completes Collector Calls.
+   */
+  private def collectScatter(collector: CollectorKey) = {
+    val shards: Iterable[CallKey] = findShards(collector)
+    val collection = for {
+      output <- generateCollectorOutput(collector, shards)
+      /* FIXME: should be replaced by awaitCallComplete(collector, output) when merging with "Database SG branch" */
+      callComplete <- awaitCallComplete(collector, output)
+    } yield callComplete
+
+    collection match {
+      case Failure(e) =>
+        /* FIXME: should be replaced by CallFailed(collector, e.getMessage) when merging with "Database SG branch" */
```

Comment:
For the 4 FIXMEs here and above, can you remove the ones that are no longer applicable?